setup.py

```diff
@@ -52,33 +52,28 @@
         "google.cloud.documentai_toolbox": ["templates/*.xml.j2"],
     },
     install_requires=(
-        "google-api-core >= 1.31.5, <3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0",
-        "pandas >= 1.0.0, <3.0.0",
-        "pandas >= 1.0.0, <2.0.0; python_version<'3.8'",
-        "pyarrow >= 15.0.0, <16.0.0; python_version>='3.8'",  # Required by Pandas #237
+        "google-api-core >= 2.17.1, <3.0.0dev,!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0",
+        "pandas[performance,gcp] >= 2.0.0, <3.0.0",
+        "pyarrow >= 15.0.0, <16.0.0",  # Required by Pandas #237
         "tabulate >= 0.9.0, <1.0.0",
-        "proto-plus >= 1.22.0, <2.0.0dev",
-        "proto-plus >= 1.22.2, <2.0.0dev; python_version>='3.11'",
+        "proto-plus >= 1.22.3, <2.0.0dev",
         "grpc-google-iam-v1 >= 0.12.6, < 0.13dev",
         "google-cloud-bigquery >= 3.5.0, < 4.0.0dev",
         "google-cloud-documentai >= 2.20.0, < 3.0.0dev",
         "google-cloud-storage >= 1.31.0, < 3.0.0dev",
         "google-cloud-vision >= 2.7.0, < 4.0.0dev",
-        "numpy >= 1.18.1, < 2.0.0",
+        "numpy >= 1.23.5, < 2.0.0",
         "intervaltree >= 3.0.0",
-        "pikepdf >= 6.2.9, < 9.0.0",
-        "pikepdf >= 6.2.9, < 7.0.0; python_version<'3.8'",
+        "pikepdf >= 8.0.0, < 9.0.0",
         "immutabledict >= 2.0.0, < 4.0.0",
-        "immutabledict >= 2.0.0, < 3.0.0dev; python_version<'3.8'",
-        "Pillow >= 9.5.0, < 11.0.0",
+        "Pillow >= 10.0.0, < 11.0.0",
         "Jinja2 >= 3.1.0, <= 4.0.0",
```

Comment:
There are a lot of changes to the minimum requirements of the client library. Can we add each of these changes to the release notes? Edit [this comment](https://github.com/googleapis/python-documentai-toolbox/pull/262#issue-2154722320) and include the following to [override the conventional commit messages](https://github.com/googleapis/release-please?tab=readme-ov-file#how-can-i-fix-release-notes) used for the release notes.
```
BEGIN_COMMIT_OVERRIDE
fix: Drop Python 3.7 Support
fix: Require google-api-core >= 2.17.1
fix: Require pandas >= 2.0.0
fix: Require proto-plus >= 1.22.3
fix: Require numpy >= 1.23.5
fix: Require pikepdf >= 8.0.0
fix: Require Pillow >= 10.0.0
END_COMMIT_OVERRIDE
```